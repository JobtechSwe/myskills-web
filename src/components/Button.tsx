import styled from '@emotion/styled'
import {
  border,
  BorderProps,
  borderRadius,
  BorderRadiusProps,
  buttonStyle,
  ButtonStyleProps,
  color,
  ColorProps,
  flex,
  FlexProps,
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  position,
  PositionProps,
  space,
  SpaceProps,
  width,
  WidthProps,
} from 'styled-system'

type VariantProps = {
  variant?: 'primary' | 'secondary' | 'secondaryBlack' | 'inActive' | 'active'
}

type ButtonProps = React.HTMLProps<HTMLButtonElement> &
  BorderProps &
  BorderRadiusProps &
  ButtonStyleProps &
  ColorProps &
  FlexProps &
  FontFamilyProps &
  FontSizeProps &
  PositionProps &
  SpaceProps &
  VariantProps &
  WidthProps

const Button = styled.button<ButtonProps>`
  cursor: pointer;
  text-decoration: none;
  white-space: nowrap;

  ${borderRadius}
  ${border}
  ${buttonStyle}
  ${color}
  ${flex}
  ${fontFamily}
  ${fontSize}
  ${position}
  ${space}
  ${width}
`

Button.defaultProps = {
  border: 'none',
  borderRadius: '8px',
  flex: ['1', 'auto'],
  fontFamily: 'default',
  fontSize: ['medium'],
  p: '12px 42px',
  variant: 'primary',
}

export const FloatingContinueButton = styled(Button)`
  position: absolute;
  bottom: 20px;
  right: 20px;
`

export default Button
