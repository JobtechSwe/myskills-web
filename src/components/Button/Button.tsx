import React from 'react'
import styled from '@emotion/styled'
import { buttonStyle } from 'styled-system'

const Btn = styled.button`
  border: 0;
  color: white;
  font-size: 20px;
  white-space: nowrap;
  text-decoration: none;
  padding: 10px 20px;
  cursor: pointer;
  ${buttonStyle}
`

interface Props {
  text: string
  variant?: string
}

const Button: React.FC<Props> = ({ text, variant }) => (
  <Btn variant={variant}>{text}</Btn>
)

export default Button
