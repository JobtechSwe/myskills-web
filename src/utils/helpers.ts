import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
