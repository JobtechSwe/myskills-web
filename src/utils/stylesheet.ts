const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
