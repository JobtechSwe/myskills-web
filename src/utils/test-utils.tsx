import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
