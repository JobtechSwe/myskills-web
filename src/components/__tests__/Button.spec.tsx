import * as React from 'react'
import Button from '../Button'
import { render } from '../../utils/test-utils'

describe('components/Button', () => {
  it('renders an Button', async () => {
    const { container } = render(<Button>Click me!</Button>)

    expect(container).toMatchSnapshot()
  })

  it('renders an Button with variant prop', async () => {
    const { container } = render(<Button variant="secondary">Click me!</Button>)

    expect(container).toMatchSnapshot()
  })
})
