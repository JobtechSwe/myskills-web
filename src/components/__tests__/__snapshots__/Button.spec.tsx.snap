// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button renders an Button 1`] = `
<div>
  <button
    class="css-w4htsj"
    font-family="default"
    font-size="medium"
  >
    Click me!
  </button>
</div>
`;

exports[`components/Button renders an Button with variant prop 1`] = `
<div>
  <button
    class="css-1g66ugz"
    font-family="default"
    font-size="medium"
  >
    Click me!
  </button>
</div>
`;
